packages/db/tests/translate/test-granules.js

```diff
@@ -217,14 +220,95 @@ test('translatePostgresGranuleToApiGranule converts Postgres granule to API gran
     ],
   };
 
-  const result = await translatePostgresGranuleToApiGranule(
-    postgresGranule,
-    knex,
+  const result = await translatePostgresGranuleToApiGranule({
+    granulePgRecord: postgresGranule,
+    knexOrTransaction: knex,
     collectionPgModel,
     pdrPgModel,
     providerPgModel,
-    filePgModel
+    filePgModel,
+  });
+
+  t.deepEqual(
+    result,
+    expectedApiGranule
   );
+});
+
+test('translatePostgresGranuleToApiGranule accepts an optional Collection', async (t) => {
```

Comment:
I'm not sure this unit is actually testing the collection logic: 

We're passing in the same collection that the granule already is referencing, and the test passes if you remove the collection. 

I think this test needs to be refactored to validate the collection passed in is the one returned, and perversely the easiest way to do that may be to pass in a *different* collection than the database. 

That, I suppose, raises an interesting question of this approach opening up the possibility that you can get a granule record that doesn't match the database in the name of query efficiency.    I'm wondering if we should validate the passed collection by the granule.collection_cumulus_id to make sure it matches?

Reply:
Ah yeah, good call. The test was basically "Does it accept the argument?" I suppose. That's not super useful. Revised to actually test that we're using the correct Collection and added validation (with another test) to ensure that the collection matches the granule. Being defensive is always a good idea.

https://github.com/nasa/cumulus/pull/2415/commits/9970cd3e6e5e79213085c6eb64a7453bef004c5d

Reply:
I think the test below might cover that (if I'm understanding correctly):

https://github.com/nasa/cumulus/pull/2415/files?file-filters%5B%5D=#diff-a4b6ca2342e9de95eded58e55b0175e246e9cc8eb56d7544d25a83180abf3070R329

The idea here is that we pass a Collection _without_ a collection_id field set. This triggers the error condition because `undefined !== the granule's collection_cumulus_id`. Eh?

Reply:
Right you are, nice work!